homeassistant/components/enphase_envoy/coordinator.py

```diff
@@ -158,6 +162,33 @@ async def _async_update_data(self) -> dict[str, Any]:
                 raise ConfigEntryAuthFailed from err
             except EnvoyError as err:
                 raise UpdateFailed(f"Error communicating with API: {err}") from err
+
+            # if we have a firmware version from previous setup, compare to current one
+            # when envoy gets new firmware there will be an authentication failure
+            # which results in getting fw version again, if so reload the integration.
+            if (current_firmware := self.envoy_firmware) and current_firmware != (
+                new_firmware := envoy.firmware
+            ):
+                _LOGGER.warning(
+                    "Envoy firmware changed from: %s to: %s, reloading enphase envoy integration",
+                    current_firmware,
+                    new_firmware,
+                )
+                persistent_notification.async_create(
+                    self.hass,
+                    title="Envoy firmware changed",
+                    message=(
+                        f"{self.name} firmware changed from: "
+                        f"{current_firmware} to: {new_firmware}, reloading integration"
+                    ),
+                    notification_id=NOTIFICATION_ID,
+                )
```

Comment:
Integrations should not create persistent notifications anymore

Reply:
removed